lib/ansible/modules/cloud/amazon/ec2_vpc_route_table_facts.py

```diff
@@ -51,79 +56,175 @@
+from ansible.module_utils.ec2 import (ec2_argument_spec, boto3_conn, ansible_dict_to_boto3_filter_list, HAS_BOTO3,
+                                      camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict)
 
-    # Add any routes to array
-    routes = []
-    associations = []
-    for route in route_table.routes:
-        routes.append(route.__dict__)
-    for association in route_table.associations:
-        associations.append(association.__dict__)
 
-    route_table_info = {'id': route_table.id,
-                        'routes': routes,
-                        'associations': associations,
-                        'tags': route_table.tags,
-                        'vpc_id': route_table.vpc_id
-                        }
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
 
-    return route_table_info
 
+def list_route_tables(route_table, module):
+    params = dict()
 
-def list_ec2_vpc_route_tables(connection, module):
-
-    filters = module.params.get("filters")
-    route_table_dict_array = []
+    params['Filters'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['RouteTableIds'] = module.params.get('route_table_ids')
 
     try:
-        all_route_tables = connection.get_all_route_tables(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
-
-    for route_table in all_route_tables:
-        route_table_dict_array.append(get_route_table_info(route_table))
-
-    module.exit_json(route_tables=route_table_dict_array)
+        result = json.loads(json.dumps(route_table.describe_route_tables(**params), default=date_handler))
+    except ValueError as e:
+        module.fail_json_aws(e, msg="Cannot validate JSON data")
+    except (ClientError, BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Could not describe route tables")
+    snaked_route_tables = [camel_dict_to_snake_dict(route_table) for route_table in result['RouteTables']]
+    if snaked_route_tables:
+        for route_table in snaked_route_tables:
+            route_table['tags'] = boto3_tag_list_to_ansible_dict(route_table.get('tags', []))
+    module.exit_json(changed=False, route_tables=snaked_route_tables)
 
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
-            filters=dict(default=None, type='dict')
+            route_table_ids=dict(default=[], type='list'),
+            filters=dict(default={}, type='dict')
         )
     )
 
-    module = AnsibleModule(argument_spec=argument_spec,
-                           supports_check_mode=True)
-
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
-
-    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              mutually_exclusive=[['vpn_connection_ids', 'filters']],
```

Comment:
s/vpn_connection_ids/route_table_ids/

But do they need to be mutually exclusive? Looks like describe_route_tables takes both at the same time: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.describe_route_tables